queasars/minimum_eigensolvers/evqe/evolutionary_algorithm/individual.py

```diff
@@ -0,0 +1,736 @@
+# Quantum Evolving Ansatz Variational Solver (QUEASARS)
+# Copyright 2023 DLR - Deutsches Zentrum für Luft- und Raumfahrt e.V.
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+from types import MappingProxyType
+from random import Random
+from math import pi, ceil
+
+from qiskit.circuit import QuantumCircuit, Parameter, Gate
+from qiskit.circuit.library import CU3Gate
+from qiskit.converters import circuit_to_gate
+
+from queasars.minimum_eigensolvers.base.evolutionary_algorithm import BaseIndividual
+from queasars.utility.random import new_random_seed
+
+
+class EVQEGateType(Enum):
+    """Enum representing the type of gate which can be placed for each qubit.
+    This class is intended for structuring the genome and not for implementing quantum gate functionality
+    """
+
+    IDENTITY = 0
+    ROTATION = 1
+    CONTROL = 2
+    CONTROLLED_ROTATION = 3
+
+
+@dataclass(frozen=True, eq=True)
+class EVQEGate(ABC):
+    """Abstract dataclass representing a quantum gate.
+    This class is intended for structuring the genome and not for implementing quantum gate functionality
```

Comment:
Yes kind of. It basically provides both an abstraction from qiskit's gate functions so that changes in qiskit are easier to handle here and it allows easier and clearer access to the qubit's indices on which the gates reside.